lib/column/array.go

```diff
@@ -191,13 +191,23 @@ func appendNullableRowPlain[T any](col *Array, arr []*T) error {
 
 func (col *Array) append(elem reflect.Value, level int) error {
 	if level < col.depth {
-		col.appendOffset(level, uint64(elem.Len()))
-		for i := 0; i < elem.Len(); i++ {
-			if err := col.append(elem.Index(i), level+1); err != nil {
-				return err
+		switch elem.Kind() {
+		// reflect.Value.Len() & reflect.Value.Index() is called in `append` method which is only valid for
+		// Slice, Array and String that make sense here.
+		case reflect.Slice, reflect.Array, reflect.String:
+			col.appendOffset(level, uint64(elem.Len()))
+			for i := 0; i < elem.Len(); i++ {
+				if err := col.append(elem.Index(i), level+1); err != nil {
+					return err
+				}
 			}
+			return nil
+		}
+		return &ColumnConverterError{
+			Op:   "AppendRow",
+			To:   "Array",
+			From: fmt.Sprintf("%T", elem),
 		}
-		return nil
```

Comment:
could you elaborate on this change please?

Reply:
While I was adding tests for the enum types it reached this code path and library panic’d on calling `elem.Len()` as the reflected value was not valid to call that method, so I added this check so at least an error would be returned instead of entire application panicing.

iirc in my case adding `Valuer` support to the Array type caused my specific tests not to hit this issue anymore, however decided to leave this safeguard as it was still possible for this panic to happen to others custom types.

Reply:
Makes sense! thanks for answering